File: 3490-find-the-maximum-length-of-valid-subsequence-i/find-the-maximum-length-of-valid-subsequence-i.py
class Solution:
    def maximumLength(self, nums: List[int]) -> int:
        k=2
        dp=[[0]*k for _ in range(k)]
        answer=0
        for num in nums:
            num%=k
            for prev in range(k):
                dp[num][prev]=dp[prev][num]+1
                answer=max(answer,dp[num][prev])
        return answer
        

File: 3490-find-the-maximum-length-of-valid-subsequence-i/find-the-maximum-length-of-valid-subsequence-i.java
class Solution {
    public int maximumLength(int[] nums) {
        int k=2;
        int[][] dp=new int[k][k];
        int answer=0;
        for(int num:nums){
            num%=k;
            for(int i=0;i<k;i++){
                dp[num][i]=dp[i][num]+1;
                answer=Math.max(answer,dp[num][i]);
            }

        }
        return answer;
        
    }
}